### PR TITLE
Drmodun/interest crud

### LINKED ISSUE #29: Interest crud

### PR DESCRIPTION
**Pull request by drmodun**
closes #29 
This pr consists of the crud endpoints for the interests resource and the schema structure of interests along with its relations

### CODE REVIEW

File: apps/api/src/interests/interests.controller.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  async create(@Body() createInterestDto: CreateInterestDto) {
+    const createdInterest =
+      await this.interestsService.create(createInterestDto);
+
+    return createdInterest;
+  }
+
+  @Get()
+  async getAll() {
+    const Interests = await this.interestsService.getAll();
```
Comment (bdeak4) [inline apps/api/src/interests/interests.controller.ts:32]: prebacit u malo slovo ime var

File: apps/api/src/interests/interests.controller.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    return Interests;
+  }
+
+  @Get(':id')
+  async findOne(@Param('id', ParseIntPipe) id: number) {
+    const Interest = await this.interestsService.getOne(id);
```
Comment (bdeak4) [inline apps/api/src/interests/interests.controller.ts:39]: prebacit u malo slovo ime var

File: packages/types/src/dto/interest/update.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { getCreateInterestDto } from './create';
+import { PartialType } from '@nestjs/mapped-types';
+
+export const getUpdateInterestDto = (ApiPropertySwagger?: any) => {
+  const CreateInterestDto = getCreateInterestDto(ApiPropertySwagger);
+
+  class UpdateInterestDto extends PartialType(CreateInterestDto) {}
```
Comment (bdeak4) [inline packages/types/src/dto/interest/update.ts:7]: isto, ne smijemo koristit ovaj partial type unutar types pa eksplicitno raspisi dto

**Review by bdeak4 — CHANGES_REQUESTED** [review]
sve u svemu LGTM, baci oko na kom

**Review by dujesa — CHANGES_REQUESTED** [review]
Sredit conflicte uz Bartolove kom

File: apps/api/src/interests/interests.controller.ts (hunk 1 above)
Comment (drmodun) [reply apps/api/src/interests/interests.controller.ts]: :white_check_mark:

File: apps/api/src/interests/interests.controller.ts (hunk 2 above)
Comment (drmodun) [reply apps/api/src/interests/interests.controller.ts]: :white_check_mark:

File: packages/types/src/dto/interest/update.ts (hunk 3 above)
Comment (drmodun) [reply packages/types/src/dto/interest/update.ts]: :white_check_mark:

**Pull request merged**